modules/heat_conduction/src/kernels/AnisoHeatConduction.C

```diff
@@ -23,53 +23,43 @@ AnisoHeatConduction::validParams()
 AnisoHeatConduction::AnisoHeatConduction(const InputParameters & parameters)
   : Kernel(parameters), _dim(_subproblem.mesh().dimension())
 {
-  _k_i[0] = _k_i[1] = _k_i[2] = NULL;
-  _k_i_dT[0] = _k_i_dT[1] = _k_i_dT[2] = NULL;
+  _k = nullptr;
+  _k_dT = nullptr;
 
-  if (hasMaterialProperty<Real>("thermal_conductivity_x"))
-    _k_i[0] = &getMaterialProperty<Real>("thermal_conductivity_x");
+  if (hasMaterialProperty<RankTwoTensor>("thermal_conductivity"))
+    _k = &getMaterialProperty<RankTwoTensor>("thermal_conductivity");
 
-  if (hasMaterialProperty<Real>("thermal_conductivity_x_dT"))
-    _k_i_dT[0] = &getMaterialProperty<Real>("thermal_conductivity_x_dT");
+  if (hasMaterialProperty<RankTwoTensor>("thermal_conductivity_dT"))
+    _k_dT = &getMaterialProperty<RankTwoTensor>("thermal_conductivity_dT");
 
-  if (hasMaterialProperty<Real>("thermal_conductivity_y"))
-    _k_i[1] = &getMaterialProperty<Real>("thermal_conductivity_y");
-
-  if (hasMaterialProperty<Real>("thermal_conductivity_y_dT"))
-    _k_i_dT[1] = &getMaterialProperty<Real>("thermal_conductivity_y_dT");
-
-  if (hasMaterialProperty<Real>("thermal_conductivity_z"))
-    _k_i[2] = &getMaterialProperty<Real>("thermal_conductivity_z");
-
-  if (hasMaterialProperty<Real>("thermal_conductivity_z_dT"))
-    _k_i_dT[2] = &getMaterialProperty<Real>("thermal_conductivity_z_dT");
-
-  if (!_k_i[0])
+  if (!_k(0,0))
     mooseError("No thermal conductivity was defined");
 
-  if (!_k_i[0] && _k_i[1])
+  if (!_k(0,0) && _k(1,1))
     mooseError("Cannot define y conductivity but not x");
```

Comment:
no, that's not how that works. Old code was checking for a null pointer, new code checks tensor entries. IMO that check should just be removed.

Reply:
Ok, revised version changes this entirely.